src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -135,8 +139,11 @@ internal void BindToDataSource(MirrorData mirrorData)
             }
             else if (this.IsExpanded)
             {
-                RefreshExpandedDisplay();
-                BeginViewSizeTransition(ComputeLargeContentSize());
+                RefreshExpandedDisplayAsync();
+
+                scheduler.ScheduleForExecution(
+                    new DelegateBasedAsyncTask(scheduler, 
+                        () => Dispatcher.BeginInvoke((Action)(() => BeginViewSizeTransition(ComputeLargeContentSize())))));
```

Comment:
I generalized this a bit.  The caller can pass in an `Action` that can be executed once `RefreshExpandedDisplay` but on the UI thread.